File: AMoAdiOSSdkDemo(Objective-C)/AMoAdiOSSdkDemo(Objective-C)/NativeHtmlViewController.m
//
//  NativeHtmlViewController.m
//  AMoAdiOSSdkDemo(Object-C)
//
//  Created by AMoAd on 2016/09/12.
//  Copyright © 2016年 AMoAd. All rights reserved.
//

#import "NativeHtmlViewController.h"
#import "AMoAdNative.h"

@interface NativeHtmlViewController ()
@end

@implementation NativeHtmlViewController
static NSString *const kTag = @"ad01";
- (void)viewDidLoad {
    [super viewDidLoad];
    [self initAd];
}

- (void)initAd {
    [AMoAdNative loadWithSid:self.sid
                         tag:kTag
                       frame:CGRectMake(100, 200, 140, 120)
                  completion:^(NSString *sid, NSString *tag, AMoAdResult result, NSDictionary *serverInfo) {
        switch (result) {
        case AMoAdResultEmpty:
            // 空の広告を受信した
            NSLog(@"空の広告を受信した");
            break;
        case AMoAdResultFailure:
            // 広告の取得に失敗した
            NSLog(@"広告の取得に失敗した");
            break;
        case AMoAdResultSuccess:
            // 正常に広告を受信した
            NSLog(@"正常に広告を受信した");
            break;
        }

    }];
    [self.view addSubview:[AMoAdNative viewWithSid:self.sid tag:kTag]];
}
- (void)showAd {
    [AMoAdNative showWithSid:self.sid tag:kTag];
}
- (void)reloadAd {
    [AMoAdNative reloadWithSid:self.sid tag:kTag];
}
- (void)hideAd {
    [AMoAdNative hideWithSid:self.sid tag:kTag];
}
- (IBAction)onShowBtnClicked:(id)sender {
    [self showAd];
}
- (IBAction)onReloadBtnClicked:(id)sender {
    [self reloadAd];
}
- (IBAction)onHideBtnClicked:(id)sender {
    [self hideAd];
}

@end

File: AMoAdiOSSdkDemo/AMoAdiOSSdkDemo/NativeHtmlViewController.swift
//
//  NativeHtmlViewController.swift
//  AMoAdiOSSdkDemo
//
//  Created by AMoAd on 2016/08/16.
//  Copyright © 2016年 AMoAd. All rights reserved.
//

import UIKit

class NativeHtmlViewController: UIViewController {
    var sid:String!
    let tag = "ad01"
    override func viewDidLoad() {
        super.viewDidLoad()
        initAd()
    }
    
    func initAd() {
        AMoAdNative.load(withSid: sid, tag: tag, frame: CGRect(x: 100, y: 200, width: 140, height: 120), completion: {(sid, tag, result, dic) in
            switch result {
            case .empty:
                // 空の広告を受信した
                break
            case .failure:
                // 広告の取得に失敗した
                break
            case .success:
                // 正常に広告を受信した
                break
            }
        })
        self.view.addSubview(AMoAdNative.view(withSid: sid, tag: tag))
    }

    func showAd() {
        AMoAdNative.show(withSid: sid, tag: tag)
    }
    
    func reloadAd() {
        AMoAdNative.reload(withSid: sid, tag: tag)
    }
    
    func hideAd() {
        AMoAdNative.hide(withSid: sid, tag:tag)
    }
    
    @IBAction func onShowBtnClicked(_ sender: AnyObject) {
        showAd()
    }
    @IBAction func onReloadBtnClicked(_ sender: AnyObject) {
        reloadAd()
    }
    @IBAction func onHideBtnClicked(_ sender: AnyObject) {
        hideAd()
    }
}
